From 292bd9426b8b3fa0308584aebf7aa9ebc87546f8 Mon Sep 17 00:00:00 2001
From: Marian 'VooDooMan' Meravy <vdm113@gmail.com>
Date: Tue, 1 Oct 2013 08:26:56 +0200
Subject: [PATCH] fix problems with ICC ("__INTELC__" macro-related)

MSVC+ICC generated plenty lots of warnings about usage of deprecated
functions in favour of ISO standard. Also, there was missing checks for
ICC macro at various places.

NB: ICC under Windows is using MSVC's headers, (even including STL), and
uses CRT libraries/DLL's from MSVC. So my patch is perfectly
functioning.

Also, disable ICC warning it emitted on each member of class saying it has
no "DLL export" while the whole class is "DLL exported".

Signed-off-by: Marian 'VooDooMan' Meravy <vdm113@gmail.com>
---
 include/wx/dlimpexp.h  | 7 ++++++-
 include/wx/wxcrtbase.h | 2 +-
 src/common/time.cpp    | 2 +-
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/include/wx/dlimpexp.h b/include/wx/dlimpexp.h
index 800b745..490b40d 100644
--- a/include/wx/dlimpexp.h
+++ b/include/wx/dlimpexp.h
@@ -23,7 +23,12 @@
        __declspec works in BC++ 5 and later, Watcom C++ 11.0 and later as well
        as VC++.
      */
-#    if defined(__VISUALC__) || defined(__BORLANDC__) || defined(__WATCOMC__)
+#    if defined(__VISUALC__) || defined(__BORLANDC__) || defined(__WATCOMC__) || defined(__VISUALC__)
+
+#    if defined(__INTELC__)
+#        pragma warning(disable: 1744) /* warning #1744: field of class type without a DLL interface used in a class with a DLL interface */
+#    endif
+
 #        define WXEXPORT __declspec(dllexport)
 #        define WXIMPORT __declspec(dllimport)
     /*
diff --git a/include/wx/wxcrtbase.h b/include/wx/wxcrtbase.h
index 79e4378..20ce182 100644
--- a/include/wx/wxcrtbase.h
+++ b/include/wx/wxcrtbase.h
@@ -165,7 +165,7 @@ WXDLLIMPEXP_BASE void *calloc( size_t num, size_t size );
 
 /* Almost all compilers have strdup(), but VC++ and MinGW call it _strdup().
    And it's not available in MinGW strict ANSI mode nor under Windows CE. */
-#if (defined(__VISUALC__) && __VISUALC__ >= 1400) || (defined(__VISUAC__) && defined (__INTELC__))
+#if (defined(__VISUALC__) && __VISUALC__ >= 1400) || (defined(_MSC_VER) && _MSC_VER >= 1400 && defined (__INTELC__))
     #define wxCRT_StrdupA _strdup
 #elif defined(__MINGW32__)
     #ifndef __WX_STRICT_ANSI_GCC__
diff --git a/src/common/time.cpp b/src/common/time.cpp
index 011b601..5c42a1d 100644
--- a/src/common/time.cpp
+++ b/src/common/time.cpp
@@ -190,7 +190,7 @@ int wxGetTimeZone()
     struct timeb tb;
     ftime(&tb);
     return tb.timezone*60;
-#elif defined(__VISUALC__)
+#elif defined(__VISUALC__) || (defined(__WINDOWS__) && defined(__INTELC__))
     // We must initialize the time zone information before using it (this will
     // be done only once internally).
     _tzset();
-- 
1.8.3.msysgit.0

